go/vt/vtgate/planbuilder/abstract/queryprojection.go

```diff
@@ -0,0 +1,206 @@
+/*
+Copyright 2021 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package abstract
+
+import (
+	"encoding/json"
+	"strconv"
+
+	vtrpcpb "vitess.io/vitess/go/vt/proto/vtrpc"
+	"vitess.io/vitess/go/vt/sqlparser"
+	"vitess.io/vitess/go/vt/vterrors"
+	"vitess.io/vitess/go/vt/vtgate/semantics"
+)
+
+type (
+	// SelectExpr provides whether the columns is aggregation expression or not.
+	SelectExpr struct {
+		Col  *sqlparser.AliasedExpr
+		Aggr bool
+	}
+
+	// QueryProjection contains the information about the projections, group by and order by expressions used to do horizon planning.
+	QueryProjection struct {
+		// If you change the contents here, please update the toString() method
+		SelectExprs        []SelectExpr
+		HasAggr            bool
+		GroupByExprs       []GroupBy
+		OrderExprs         []OrderBy
+		CanPushDownSorting bool
```

Comment:
hasAggr and CanPushDownSorting should be added in toString method and a test can be added as well.